### PR TITLE
New version of dendrogram

### PR DESCRIPTION
**Pull request by yihming**
* Use Connection Specific Index (CSI) for the distance calculation;
* Split into 2 functions: 
  * `calc_dendrogram` to calculate linkage, then store in `data.uns`;
  * `plot_dendrogram` to plot a dendrogram based on the linkage calculated.

### CODE REVIEW

File: pegasus/plotting/plot_library.py [hunk 1]
```diff
@@ -1581,6 +1582,108 @@ def non_zero(g):
...
+        color_threshold=color_threshold,
+        orientation=orientation,
+    )
+    fig.tight_layout()
+    plt.xticks(rotation=label_rotation)
+
+    return fig if return_fig else None
+
+
+def _dendrogram_obsolete(
```
Comment (bli25) [inline pegasus/plotting/plot_library.py:1654]: Should we delete the obsoleted function here?

Comment (yihming) [reply pegasus/plotting/plot_library.py:1654]: Yes. When this PR is ready to merge, I'll delete it. I keep it here just for testing purpose.

File: pegasus/plotting/plot_library.py [hunk 2]
```diff
@@ -1641,44 +1630,23 @@ def dendrogram(
     >>> pg.dendrogram(data, rep='pca', groupby='leiden_labels')
```
Comment (bli25) [inline pegasus/plotting/plot_library.py]: Please change the example here.

**Review by bli25 — CHANGES_REQUESTED** [review]
Please address the minor comment and feel free to merge!

**Pull request merged**